src/charm.py

```diff
@@ -155,23 +159,166 @@ def _on_munge_available(self, event):
+                    continue
+
+                keys_processed.add(var_name)
+                # If explicitly None, then remove from the output buffer as
+                # the demand is to unset the variable.
+                if kwargs[var_name] is None:
+                    continue
+
+                updated_contents.append(f"{var_name.upper()}={kwargs[var_name]}\n")
+
+            for key, value in kwargs.items():
+                # Skip any keys already processed.
+                if key in keys_processed:
+                    continue
+                updated_contents.append(f"{key.upper()}={value}\n")
+
+            f.seek(0)
+            f.truncate()
+            f.writelines(updated_contents)
+
     def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
-        """Set database info in StoredState and invoke _write_config_and_restart_slurmdbd.
+        """Process the DatabaseCreatedEvent and updates the database parameters.
+
+        Updates the database parameters for the slurmdbd configuration based up on the
+        DatabaseCreatedEvent. The type of update depends on the endpoints provided in
+        the DatabaseCreatedEvent.
+
+        If the endpoints provided are file paths to unix sockets
+        then the /etc/default/slurmdbd file will be updated to tell the MySQL client to
+        use the socket.
+
+        If the endpoints provided are Address:Port tuples, then the address and port are
+        updated as the database parameters in the slurmdbd.conf configuration file.
 
         Args:
             event (DatabaseCreatedEvent):
                 Information passed by MySQL after the slurm_acct_db database has been created.
+
+        Raises:
+            ValueError:
+                When the database endpoints are invalid (e.g. empty).
         """
         logger.debug("Configuring new backend database for slurmdbd.")
-        self.set_db_info(
-            {
-                "db_username": event.username,
-                "db_password": event.password,
-                "db_hostname": "127.0.0.1",
-                "db_port": "3306",
-                "db_name": "slurm_acct_db",
-            }
-        )
+
+        socket_endpoints = []
+        tcp_endpoints = []
+        if not event.endpoints:
+            # This is 100% an error condition that the charm doesn't know how to handle
+            # and is an unexpected condition. Raise an error here to fail the hook in
+            # a bad way. The event isn't deferred as this is a situation that requires
+            # a human to look at and resolve the proper next steps. Reprocessing the
+            # deferred event will only result in continual errors.
+            logger.error(f"No endpoints provided: {event.endpoints}")
+            self.unit.status = ErrorStatus("No database endpoints")
+            raise ValueError(f"Unexpected endpoint types: {event.endpoints}")
+
+        for endpoint in [ep.strip() for ep in event.endpoints.split(",")]:
```

Comment:
The strip is a safety check for formatting changes. `Foo, Bar` is still comma separated but with whitespace and it will result in `['Foo', ' Bar']` using `str.split(",")` which is incorrect for the usage.